app/src/main/java/com/getyourguide/interview/ReviewsFragment.kt

```diff
@@ -4,56 +4,179 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
 import androidx.fragment.app.Fragment
-import androidx.recyclerview.widget.LinearLayoutManager
-import com.getyourguide.interview.databinding.FragmentReviewsBinding
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.flow.flow
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
+import coil.compose.AsyncImage
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
 
 class ReviewsFragment : Fragment() {
 
   companion object {
     fun newInstance(): Fragment = ReviewsFragment()
   }
 
-  private var _binding: FragmentReviewsBinding? = null
-  private val binding: FragmentReviewsBinding
-    get() = _binding!!
-
   private val api: ReviewsApi = RetrofitBuilder.build(ReviewsApi::class.java)
-  private val adapter = ReviewsAdapter()
 
-  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
-    _binding = FragmentReviewsBinding.inflate(inflater, container, false)
-    return binding.root
-  }
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View {
+    return ComposeView(requireContext()).apply {
+      setViewCompositionStrategy(
+        ViewCompositionStrategy.DisposeOnLifecycleDestroyed(
+          viewLifecycleOwner
+        )
+      )
```

Comment:
Is this needed for the interview?

Reply:
no But it will create a leak i think, should i remove it?

Reply:
I don't think it will create a leak, as it will be `ViewCompositionStrategy.Default` if we don't set anything.

Reply:
yeah just saw the code base, should i remove setViewCompositionStrategy all together then as by default its alredy set to dispose the compose when the view is detached from window.

Reply:
Yeah, I would remove this as it can confuse the candidate and I don't think having it's of the highest importance.